src/project/yaml_provider.rs

```diff
@@ -1,9 +1,85 @@
 use chrono::prelude::*;
 use yaml_rust::{Yaml, yaml::Hash as YamlHash};
 
-use crate::util::yaml::parse_dmy_date;
+use crate::util::yaml::{parse_dmy_date, parse_dmy_date_range};
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum FieldValue<T> {
```

Comment:
so, do we want `Result<E, T>` with custom error type `E` or `Result<E, Option<T>>`? 

Using `Result`, we need to be careful what we do when there are multiple alternatives for a field (for example `event.name` or just `event`). Do we want to fallback on alternatives if there is invalid data (ignoring the error)? IMO, we only want the fallback if data is actually missing, and other errors should just produce an error.

Reply:
What you have is still the `Ok` vs `Err` scenario, but `Err` can be because it's `missing` or because it's `nvalid`. Have a look here: https://github.com/ascii-dresden/asciii/tree/feature/FieldResult

Reply:
But then this code:

https://github.com/ascii-dresden/asciii/blob/ef5d5af27cd41dd489ba77566444359ba6e585c9/src/project/spec_yaml.rs#L192-L196

is wrong, since it ignores invalid data errors in `"invoice.payed_date"`. But perhaps we actually want to check that only one of the alternatives exists, because silently choosing one of multiple alternatives also seems a bit unintuitive and may cause hard to understand behaviour.

Reply:
this is only there because there is still that old format from 5 years that needs to work